### PR TITLE
Qa/kafka feature

### PR DESCRIPTION
**Pull request by steel4ge**
Pull request provides the possibility to customise flexible procedure for kafka brocker installation.

### CODE REVIEW

File: roles/kafka-mesos/defaults/main.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ---
+enable_client_install: false
+destroy_kafka_brockers: false
```
Comment (steel4ge) [inline roles/kafka-mesos/defaults/main.yml]: Should be renamed to repair

File: roles/kafka-mesos/README.rst [hunk 2]
```diff
@@ -5,7 +5,8 @@ Kafka Mesos
 
 Variables
 ---------
-
+- `destroy_kafka_brockers: enable or disable kafka destruction of all brockers on node(s). Default: `false`
```
Comment (OleksandrBerchenko) [inline roles/kafka-mesos/README.rst]: repair

File: roles/kafka-mesos/defaults/main.yml [hunk 3]
```diff
@@ -1,4 +1,6 @@
 ---
+enable_client_install: false
```
Comment (OleksandrBerchenko) [inline roles/kafka-mesos/defaults/main.yml]: Can you remind me why it's "false" by default?

File: roles/kafka-mesos/tasks/destroy.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+---
+- name: Register running brockers
```
Comment (OleksandrBerchenko) [inline roles/kafka-mesos/tasks/destroy.yml]: Calculate running brockers

File: roles/kafka-mesos/tasks/destroy.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  args:
+    chdir: "{{ kafka_mesos_install_dir }}"
+  tags:
+    - kafka-mesos
+  failed_when: existing_brockers.stdout == -1
+
+- name: Stop all Kafka brockers
+  run_once: true
+  sudo: true
+  command: "./kafka-mesos.sh stop 0..{{ existing_brockers.stdout | int - 1 }} --api http://{{ kafka_scheduler_hostname }}:7000"
```
Comment (OleksandrBerchenko) [inline roles/kafka-mesos/tasks/destroy.yml]: 1. What if there are 0 brockers?  "stop 0..-1"?  Don't you need "when" condition?
2. What if there is 1 brocker?  Is "0..0" a correct syntax?  Did you test it?

File: roles/kafka-mesos/tasks/destroy.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  args:
+    chdir: "{{ kafka_mesos_install_dir }}"
+  tags:
+    - kafka-mesos
+
+- name: Remove all Kafka brockers
+  run_once: true
+  sudo: true
+  command: "./kafka-mesos.sh remove 0..{{ existing_brockers.stdout | int - 1 }} --heap {{ kafka_broker_heap }} --mem {{ kafka_broker_memory }} --api http://{{ kafka_scheduler_hostname }}:7000"
```
Comment (OleksandrBerchenko) [inline roles/kafka-mesos/tasks/destroy.yml]: See above.

File: roles/kafka-mesos/tasks/client.yml [hunk 7]
```diff
@@ -1,27 +1,48 @@
 ---
-- name: Wait for Kafka scheduler
+- name: Register running brockers
```
Comment (OleksandrBerchenko) [inline roles/kafka-mesos/tasks/client.yml]: Calculate running brockers

File: roles/kafka-mesos/tasks/client.yml [hunk 8]
```diff
@@ -1,27 +1,48 @@
...
+  shell: ./kafka-mesos.sh status | grep -c "id:.[0-9]"
+  register: existing_brockers
+  args:
+    chdir: "{{ kafka_mesos_install_dir }}"
+  tags:
+    - kafka-mesos
+  failed_when: existing_brockers.stdout == -1
+
+- name: Additional Kafka brockers
```
Comment (OleksandrBerchenko) [inline roles/kafka-mesos/tasks/client.yml]: Add..

File: roles/kafka-mesos/tasks/client.yml [hunk 9]
```diff
@@ -1,27 +1,48 @@
...
+  when: "{{ existing_brockers.stdout | int }} < {{ kafka_brokers }}"
   run_once: true
-  wait_for:
-    host: "{{ ansible_hostname }}"
-    port: 7000
-    timeout: 1800
-    delay: 20
+  sudo: true
+  command: './kafka-mesos.sh add {{ existing_brockers.stdout }}..{{ kafka_brokers - 1 }} --heap {{ kafka_broker_heap }} --mem {{ kafka_broker_memory }} --api http://{{ kafka_scheduler_hostname }}:7000'
```
Comment (OleksandrBerchenko) [inline roles/kafka-mesos/tasks/client.yml]: Is something like "add 5..5" a correct syntax?

File: roles/kafka-mesos/tasks/main.yml [hunk 10]
```diff
@@ -42,10 +34,26 @@
...
+- name: Create Kafka-Mesos configuration
+  sudo: true
+  template:
+    src: "kafka.properties.j2"
+    dest: "{{ kafka_mesos_install_dir }}/kafka-mesos.properties"
+  tags:
+    - kafka-mesos
+
+- include: destroy.yml
+  when: destroy_kafka_brockers == true
```
Comment (OleksandrBerchenko) [inline roles/kafka-mesos/tasks/main.yml]: repair?  How did you test the changes?

File: roles/kafka-mesos/README.rst [hunk 11]
```diff
@@ -5,7 +5,8 @@ Kafka Mesos
 
 Variables
 ---------
-
+- `repair: enable or disable kafka destruction of all brockers on node(s). Default: `false`
+- `enable_client_install`: enable or disable kafka brockers installation on node(s). Default: `false`
```
Comment (OleksandrBerchenko) [inline roles/kafka-mesos/README.rst]: true

**Comment by ohal** [comment]
pls, take a look:
```
TASK: [kafka-mesos | Download kafka-mesos repo] ******************************* <10.203.30.75> ESTABLISH CONNECTION FOR USER: centos <10.203.30.75> REMOTE_MODULE git dest=/opt/kafka-mesos repo=https://github.com/mesos/kafka
<10.203.30.75> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.30.75 /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1437641872.35-274054316942460 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1437641872.35-274054316942460 && echo $HOME/.ansible/tmp/ansible-tmp-1437641872.35-274054316942460'
<10.203.30.75> PUT /tmp/tmpSVBYtP TO /home/centos/.ansible/tmp/ansible-tmp-1437641872.35-274054316942460/git
<10.203.30.75> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.30.75 /bin/sh -c 'sudo -k && sudo -H -S -p "[sudo via ansible, key=fvxrrxqqpdsaiwrbylwoqxzehpktgpvw] password: " -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-fvxrrxqqpdsaiwrbylwoqxzehpktgpvw; LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /home/centos/.ansible/tmp/ansible-tmp-1437641872.35-274054316942460/git; rm -rf /home/centos/.ansible/tmp/ansible-tmp-1437641872.35-274054316942460/ >/dev/null 2>&1'"'"''
failed: [host-04] => {"failed": true}
msg: Failed to find required executable git <10.203.28.14> ESTABLISH CONNECTION FOR USER: centos <10.203.28.14> REMOTE_MODULE git dest=/opt/kafka-mesos repo=https://github.com/mesos/kafka
<10.203.28.14> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.28.14 /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1437641872.36-237710210525102 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1437641872.36-237710210525102 && echo $HOME/.ansible/tmp/ansible-tmp-1437641872.36-237710210525102'
<10.203.28.14> PUT /tmp/tmpqnrMg7 TO /home/centos/.ansible/tmp/ansible-tmp-1437641872.36-237710210525102/git
<10.203.28.14> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.28.14 /bin/sh -c 'sudo -k && sudo -H -S -p "[sudo via ansible, key=exvtfpppfoesyhdgtfrgazkeptivkrcb] password: " -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-exvtfpppfoesyhdgtfrgazkeptivkrcb; LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /home/centos/.ansible/tmp/ansible-tmp-1437641872.36-237710210525102/git; rm -rf /home/centos/.ansible/tmp/ansible-tmp-1437641872.36-237710210525102/ >/dev/null 2>&1'"'"''
failed: [host-05] => {"failed": true}
msg: Failed to find required executable git <10.203.28.146> ESTABLISH CONNECTION FOR USER: centos <10.203.28.146> REMOTE_MODULE git dest=/opt/kafka-mesos repo=https://github.com/mesos/kafka
<10.203.28.146> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.28.146 /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1437641872.39-260655198646821 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1437641872.39-260655198646821 && echo $HOME/.ansible/tmp/ansible-tmp-1437641872.39-260655198646821'
<10.203.28.146> PUT /tmp/tmpIGXWNs TO /home/centos/.ansible/tmp/ansible-tmp-1437641872.39-260655198646821/git
<10.203.28.146> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.28.146 /bin/sh -c 'sudo -k && sudo -H -S -p "[sudo via ansible, key=cmcpnyrgrbsaltotaoqcjevzoxhqtvvx] password: " -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-cmcpnyrgrbsaltotaoqcjevzoxhqtvvx; LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /home/centos/.ansible/tmp/ansible-tmp-1437641872.39-260655198646821/git; rm -rf /home/centos/.ansible/tmp/ansible-tmp-1437641872.39-260655198646821/ >/dev/null 2>&1'"'"''
failed: [host-06] => {"failed": true}
msg: Failed to find required executable git <10.203.30.69> ESTABLISH CONNECTION FOR USER: centos <10.203.30.69> REMOTE_MODULE git dest=/opt/kafka-mesos repo=https://github.com/mesos/kafka
<10.203.30.69> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.30.69 /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1437641872.41-89124255955294 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1437641872.41-89124255955294 && echo $HOME/.ansible/tmp/ansible-tmp-1437641872.41-89124255955294'
<10.203.30.69> PUT /tmp/tmp5YQKmD TO /home/centos/.ansible/tmp/ansible-tmp-1437641872.41-89124255955294/git
<10.203.30.69> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.30.69 /bin/sh -c 'sudo -k && sudo -H -S -p "[sudo via ansible, key=fyufkgqniwsunciavwkrwbcdspjjbzng] password: " -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-fyufkgqniwsunciavwkrwbcdspjjbzng; LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /home/centos/.ansible/tmp/ansible-tmp-1437641872.41-89124255955294/git; rm -rf /home/centos/.ansible/tmp/ansible-tmp-1437641872.41-89124255955294/ >/dev/null 2>&1'"'"''
failed: [host-07] => {"failed": true}
msg: Failed to find required executable git

```

**Comment by ohal** [comment]
another issue is:
```
NOTIFIED: [marathon | wait for marathon to listen] **************************** 
<10.203.28.248> ESTABLISH CONNECTION FOR USER: centos
<10.203.28.248> REMOTE_MODULE command /usr/local/bin/marathon-wait-for-listen.sh
<10.203.28.248> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.28.248 /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1437646990.61-264925769276859 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1437646990.61-264925769276859 && echo $HOME/.ansible/tmp/ansible-tmp-1437646990.61-264925769276859'
<10.203.28.248> PUT /tmp/tmppyPCMx TO /home/centos/.ansible/tmp/ansible-tmp-1437646990.61-264925769276859/command
<10.203.28.248> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.28.248 /bin/sh -c 'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /home/centos/.ansible/tmp/ansible-tmp-1437646990.61-264925769276859/command; rm -rf /home/centos/.ansible/tmp/ansible-tmp-1437646990.61-264925769276859/ >/dev/null 2>&1'
changed: [host-02] => {"changed": true, "cmd": ["/usr/local/bin/marathon-wait-for-listen.sh"], "delta": "0:00:03.779303", "end": "2015-07-23 10:23:14.612210", "rc": 0, "start": "2015-07-23 10:23:10.832907", "stderr": "", "stdout": "", "warnings": []}
<10.203.28.33> ESTABLISH CONNECTION FOR USER: centos
<10.203.28.33> REMOTE_MODULE command /usr/local/bin/marathon-wait-for-listen.sh
<10.203.28.33> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.28.33 /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1437646990.59-131163935716212 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1437646990.59-131163935716212 && echo $HOME/.ansible/tmp/ansible-tmp-1437646990.59-131163935716212'
<10.203.28.33> PUT /tmp/tmpHk1Z1l TO /home/centos/.ansible/tmp/ansible-tmp-1437646990.59-131163935716212/command
<10.203.28.33> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.28.33 /bin/sh -c 'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /home/centos/.ansible/tmp/ansible-tmp-1437646990.59-131163935716212/command; rm -rf /home/centos/.ansible/tmp/ansible-tmp-1437646990.59-131163935716212/ >/dev/null 2>&1'
changed: [host-01] => {"changed": true, "cmd": ["/usr/local/bin/marathon-wait-for-listen.sh"], "delta": "0:00:03.837549", "end": "2015-07-23 10:23:14.649456", "rc": 0, "start": "2015-07-23 10:23:10.811907", "stderr": "", "stdout": "", "warnings": []}
<10.203.29.120> ESTABLISH CONNECTION FOR USER: centos
<10.203.29.120> REMOTE_MODULE command /usr/local/bin/marathon-wait-for-listen.sh
<10.203.29.120> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.29.120 /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1437646990.62-15173725960514 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1437646990.62-15173725960514 && echo $HOME/.ansible/tmp/ansible-tmp-1437646990.62-15173725960514'
<10.203.29.120> PUT /tmp/tmpfDAXLk TO /home/centos/.ansible/tmp/ansible-tmp-1437646990.62-15173725960514/command
<10.203.29.120> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.29.120 /bin/sh -c 'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /home/centos/.ansible/tmp/ansible-tmp-1437646990.62-15173725960514/command; rm -rf /home/centos/.ansible/tmp/ansible-tmp-1437646990.62-15173725960514/ >/dev/null 2>&1'
failed: [host-03] => {"changed": true, "cmd": ["/usr/local/bin/marathon-wait-for-listen.sh"], "delta": "0:01:00.257752", "end": "2015-07-23 10:24:11.090868", "rc": 2, "start": "2015-07-23 10:23:10.833116", "warnings": []}

```

**Comment by ohal** [comment]
next one is:
```
TASK: [spark | Download Spark binaries] *************************************** 
<10.203.25.159> ESTABLISH CONNECTION FOR USER: centos
<10.203.25.159> REMOTE_MODULE get_url dest=/opt/spark-1.3.1-bin-hadoop2.6.tgz url=http://mirror.ox.ac.uk/sites/rsync.apache.org/spark/spark-1.3.1/spark-1.3.1-bin-hadoop2.6.tgz
<10.203.25.159> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.25.159 /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1437652039.67-248550338555718 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1437652039.67-248550338555718 && echo $HOME/.ansible/tmp/ansible-tmp-1437652039.67-248550338555718'
<10.203.25.159> PUT /tmp/tmp_EyXTS TO /home/centos/.ansible/tmp/ansible-tmp-1437652039.67-248550338555718/get_url
<10.203.25.159> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/jenkins/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 10.203.25.159 /bin/sh -c 'sudo -k && sudo -H -S -p "[sudo via ansible, key=jzkgtfirmnlxgxtaluvuxbamzbsppvfd] password: " -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-jzkgtfirmnlxgxtaluvuxbamzbsppvfd; LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /home/centos/.ansible/tmp/ansible-tmp-1437652039.67-248550338555718/get_url; rm -rf /home/centos/.ansible/tmp/ansible-tmp-1437652039.67-248550338555718/ >/dev/null 2>&1'"'"''
failed: [host-08] => (item=spark-1.3.1-bin-hadoop2.6) => {"dest": "/opt/spark-1.3.1-bin-hadoop2.6.tgz", "failed": true, "item": "spark-1.3.1-bin-hadoop2.6", "response": "Connection failure: timed out", "state": "absent", "status_code": -1, "url": "http://mirror.ox.ac.uk/sites/rsync.apache.org/spark/spark-1.3.1/spark-1.3.1-bin-hadoop2.6.tgz"}
msg: Request failed
```

**Pull request merged**